pkg/controller/importconfig/cluster_info.go

```diff
@@ -144,3 +109,183 @@ func validateToken(token string, creation, expiration []byte) bool {
 	lifetime := time.Until(expirationTime)
 	return lifetime > refreshThreshold
 }
+
+func buildBootstrapKubeconfigData(ctx context.Context, clientHolder *helpers.ClientHolder,
+	managedCluster *clusterv1.ManagedCluster,
+	klusterletConfig *klusterletconfigv1alpha1.KlusterletConfig) ([]byte, []byte, []byte, error) {
+	var bootstrapKubeconfigData, tokenData, tokenCreation, tokenExpiration []byte
+
+	// get the import secret
+	importSecret, err := getImportSecret(ctx, clientHolder, managedCluster.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, nil, nil, err
+	}
+
+	// get the latest kube apiserver configuration
+	requiredKubeAPIServer, requiredProxyURL, requiredCA, requiredCAData, err := bootstrap.GetKubeAPIServerConfig(
+		ctx, clientHolder, managedCluster.Name, klusterletConfig, isSelfManaged(managedCluster))
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// get the cluster name in the kubeconfig
+	requiredCtxClusterName, err := bootstrap.GetKubeconfigClusterName(ctx, clientHolder.RuntimeClient)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if importSecret == nil {
+		klog.Infof("the import secret is missing for the managed cluster %s", managedCluster.Name)
+	}
+
+	// check if the bootstrap kubeconfig and token in the import secret are still valid
+	if kubeconfigData := extractBootstrapKubeConfigDataFromImportSecret(importSecret); len(kubeconfigData) > 0 {
+		kubeAPIServer, proxyURL, ca, caData, tokenString, ctxClusterName, err := parseKubeConfigData(kubeconfigData)
+		if err != nil {
+			klog.Errorf("failed to parse the bootstrap hub kubeconfig in the import.yaml. Recreation is required: %v", err)
```

Comment:
why do not we return here?

Reply:
It will result in an infinite loop if an error is returned when the existing kubeconfig is invalid. We'd better to log the error and regenerate the bootstrap kubeconfig in this case.

Reply:
Changed the log level to `Info`.